pinot-common/src/main/java/org/apache/pinot/common/function/FunctionInvoker.java

```diff
@@ -112,7 +113,12 @@ public void convertTypes(Object[] arguments) {
       PinotDataType argumentType = FunctionUtils.getArgumentType(argumentClass);
       Preconditions.checkArgument(parameterType != null && argumentType != null,
           "Cannot convert value from class: %s to class: %s", argumentClass, parameterClass);
-      arguments[i] = parameterType.convert(argument, argumentType);
+      try {
+        arguments[i] = parameterType.convert(argument, argumentType);
+      } catch (Exception e) {
+        String errorMsg = "Invalid conversion when calling " + _method + ": " + e.getMessage();
+        throw QueryErrorCode.QUERY_EXECUTION.asException(errorMsg, e);
+      }
```

Comment:
This is called in various ingestion paths as well, I don't think we should be using query execution error here.

Reply:
AFAIK, when ingestion calls this code, what they are using is to evaluate some built-in functions to apply transformations. I think it makes sense to receive query exceptions in that case. Don't you?

Reply:
I'm not sure I agree that query processing related exceptions should ever be used in ingestion related error paths even if function related (since those are common components). But I'd prefer to not block this PR any longer, so we can discuss this further separately.